lib/gs-plugin-loader.c

```diff
@@ -1797,23 +1775,18 @@ static void
 save_install_queue (GsPluginLoader *plugin_loader)
 {
 	GsPluginLoaderPrivate *priv = gs_plugin_loader_get_instance_private (plugin_loader);
-	GPtrArray *pending_apps;
 	gboolean ret;
 	gint i;
 	g_autoptr(GError) error = NULL;
 	g_autoptr(GString) s = NULL;
 	g_autofree gchar *file = NULL;
 
 	s = g_string_new ("");
-	pending_apps = priv->pending_apps;
 	g_mutex_lock (&priv->pending_apps_mutex);
-	for (i = (gint) pending_apps->len - 1; i >= 0; i--) {
-		GsApp *app;
-		app = g_ptr_array_index (pending_apps, i);
-		if (gs_app_get_state (app) == AS_APP_STATE_QUEUED_FOR_INSTALL) {
```

Comment:
Is it correct that this check for AS_APP_STATE_QUEUED_FOR_INSTALL is no longer necessary?

Reply:
This question still stands, at least with regard to commit https://github.com/endlessm/gnome-software/pull/326/commits/27a5e769746c7ac2dd641f3b01e55d5708dadcb0

Reply:
Honestly, since the commit is months old already, I don't remember exactly. I think there were several problems related to the apps not being in this state while dealing with them, and decoupling the state from the queue means we don't have to rely on the state (one less way to get the typical issues we have with g-s).

I'd have to thoroughly retest all this again to give you a better answer (let me know if I should do it).

Reply:
Okay, as long as you think it was intentional, I'm fine.